contracts/strategies/notional/NotionalStrategy.sol

```diff
@@ -0,0 +1,297 @@
+        (, Types.Token memory underlyingToken) = NotionalProxy(_rewardPool).getCurrency(_currencyId);
+
+        _setCurrencyId(bytes32(uint256(_currencyId)));
+        _setBVault(_bVault);
+        _setNote2WethPoolId(_note2wethpid);
+        _setNTokenUnderlying(underlyingToken.tokenAddress);
+    }
+
+    /* ========== View ========== */
+
+    function depositArbCheck() public view returns (bool) {
+        return true;
+    }
+
+    function unsalvagableTokens(address token) public view returns (bool) {
+        return (token == rewardToken() || token == underlying());
+    }
+
+    function investedUnderlyingBalance() public view returns (uint256) {
+        uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+
+        return underlyingBalance;
+    }
+
+    /* ========== Internal ========== */
+
+    function _mintNTokens() internal {
+        address _nTokenUnderlying = nTokenUnderlying();
+        address _rewardPool = rewardPool();
+
+        uint256 underlyingBalance;
+
+        if (_nTokenUnderlying == address(0)) {
+            underlyingBalance = IERC20(weth).balanceOf(address(this));
+            WETH9(weth).withdraw(underlyingBalance);
+        } else {
+            underlyingBalance = IERC20(_nTokenUnderlying).balanceOf(address(this));
+            IERC20(_nTokenUnderlying).safeApprove(_rewardPool, 0);
+            IERC20(_nTokenUnderlying).safeApprove(_rewardPool, underlyingBalance);
+        }
+
+        Types.BalanceAction memory action = Types.BalanceAction(
+            Types.DepositActionType.DepositUnderlyingAndMintNToken,
+            currencyId(),
+            underlyingBalance,
+            0,
+            false,
+            false
+        );
+        Types.BalanceAction[] memory actions = new Types.BalanceAction[](1);
+        actions[0] = action;
+
+        if (_nTokenUnderlying != address(0)) {
+            underlyingBalance = 0;
+        }
+
+        NotionalProxy(_rewardPool).batchBalanceAction.value(underlyingBalance)(address(this), actions);
+    }
+
+    function _claimRewards() internal {
+        NotionalProxy(rewardPool()).nTokenClaimIncentives();
+    }
+
+    function _note2WETH() internal {
+        uint256 noteAmount = IERC20(note).balanceOf(address(this));
+
+        if (noteAmount == 0) {
+            return;
+        }
+
+        //swap note to weth on balancer
+        IBVault.SingleSwap memory singleSwap;
+        IBVault.SwapKind swapKind = IBVault.SwapKind.GIVEN_IN;
+
+        singleSwap.poolId = note2WethPoolId();
+        singleSwap.kind = swapKind;
+        singleSwap.assetIn = IAsset(note);
+        singleSwap.assetOut = IAsset(weth);
+        singleSwap.amount = noteAmount;
+        singleSwap.userData = abi.encode(0);
+
+        IBVault.FundManagement memory funds;
+        funds.sender = address(this);
+        funds.fromInternalBalance = false;
+        funds.recipient = address(uint160(address(this)));
```

Comment:
is it necessary wrapping?

Reply:
We need it to cast `address` to `payable address`.